File: 777-toeplitz-matrix/toeplitz-matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix):
        rows = len(matrix)
        cols = len(matrix[0])

        for i in range(1, rows):
            for j in range(1, cols):

                if matrix[i][j] != matrix[i - 1][j - 1]:
                    return False

        return True

File: 777-toeplitz-matrix/toeplitz-matrix.java
class Solution {
    public boolean isToeplitzMatrix(int[][] matrix) {

        int rows = matrix.length;
        int cols = matrix[0].length;

        for (int i = 1; i < rows; i++) {
            for (int j = 1; j < cols; j++) {

                if (matrix[i][j] != matrix[i - 1][j - 1]) {
                    return false;
                }
            }
        }

        return true;
    }
}